superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -171,8 +200,24 @@ export default class ResultSet extends React.PureComponent<
       appContainer?.getAttribute('data-bootstrap') || '{}',
     );
 
-    const datasets = await getByUser(bootstrapData.user.userId);
-    const userDatasetsOwned = datasets.map(
+    const queryParams = rison.encode({
+      filters: [
+        {
+          col: 'owners',
+          opr: 'rel_m_m',
+          value: bootstrapData.user.userId,
+        },
+      ],
+      order_column: 'changed_on_delta_humanized',
+      order_direction: 'desc',
+    });
+
+    const response = await makeApi({
+      method: 'GET',
+      endpoint: `/api/v1/dataset?q=${queryParams}`,
+    })({});
+
+    const userDatasetsOwned = response.result.map(
```

Comment:
could you get a response in a format other than {result: []}?

Reply:
I think anything accept a 200 would break this code thinking I could put a guard statement here